casper/src/test/scala/coop/rchain/casper/sync/BlockRetrieverRequesAllSpec.scala

```diff
@@ -0,0 +1,199 @@
+    BlockRequest.from(convert[PacketTypeTag.BlockRequest.type](toPacket(protocol).right.get).get)
+  private def toHasBlockRequest(protocol: Protocol): HasBlockRequest =
+    HasBlockRequest.from(
+      convert[PacketTypeTag.HasBlockRequest.type](toPacket(protocol).right.get).get
+    )
+
+  private def endpoint(port: Int): Endpoint = Endpoint("host", port, port)
+  private def peerNode(name: String, port: Int = 40400): PeerNode =
+    PeerNode(NodeIdentifier(name.getBytes), endpoint(port))
+
+  private def alwaysSuccess: PeerNode => Protocol => CommErr[Unit] = kp(kp(Right(())))
+
+  private def timedOut: Long    = time.clock - (2 * timeout.toMillis)
+  private def notTimedOut: Long = time.clock - 1
+
+  override def beforeEach(): Unit = {
+    transportLayer.reset()
+    transportLayer.setResponses(alwaysSuccess)
+    log.reset()
+    time.reset()
+  }
+
+  describe("Running") {
+    describe("maintainRequestedBlocks, for every block that was requested") {
+      describe("if block request is still within a timeout") {
+        it("should keep the request not touch") {
+          val requested = RequestState(timestamp = notTimedOut)
+          currentRequests.set(Map(hash -> requested)).runSyncUnsafe()
+          // when
+          blockRetriever.requestAll(timeout).runSyncUnsafe()
+          // then
+          val requestedBlocksMapAfter = currentRequests.get.runSyncUnsafe()
+          requestedBlocksMapAfter.size should be(1)
+        }
+      }
+      describe("if block was not delivered within given timeout") {
+        describe("if waiting list is not empty") {
+          it("should request block from first peer on a waiting list") {
+            // given
+            val waitingList = List(peerNode("waiting1"), peerNode("waiting2"))
+            val requested = RequestState(
+              timestamp = timedOut,
+              peers = Set(peerNode("peer")),
+              waitingList = waitingList
+            )
+            currentRequests.set(Map(hash -> requested)).runSyncUnsafe()
+            // when
+            blockRetriever.requestAll(timeout).runSyncUnsafe()
+            // then
+            val (recipient, msg) = transportLayer.getRequest(0)
+            toBlockRequest(msg).hash should be(hash)
+            recipient shouldBe waitingList.head
+            transportLayer.requests.size shouldBe 1
+          }
+          it("should move that peer from the waiting list to the requested set") {
+            // given
+            val waitingList = List(peerNode("waiting1"), peerNode("waiting2"))
+            val requested = RequestState(
+              timestamp = timedOut,
+              peers = Set(peerNode("peer")),
+              waitingList = waitingList
+            )
+            currentRequests.set(Map(hash -> requested)).runSyncUnsafe()
+            // when
+            blockRetriever.requestAll(timeout).runSyncUnsafe()
+            // then
+            val Some(requestedAfter) = currentRequests.get.runSyncUnsafe().get(hash)
```

Comment:
Just want to say that you can wrap  these into a `for` and then you don't have to call `runSyncUnsafe()` every time.

Reply:
Ah, I see, that makes sense